plone/app/z3cform/widget.py

```diff
@@ -619,6 +620,34 @@ def _base_args(self):
 
         return args
 
+    def items(self):
+        """Return item for the widget values for the display template
+
+        Query the catalog for the widget-value (uuids) to only display items
+        that the user is allowed to see. Accessing the value with e.g.
+        getattr(self.context, self.__name__) would yield the items unfiltered.
+        To keep the order intact query each uuid individually.
+        Uses IContentListing for easy access to MimeTypeIcon and more.
+        """
+        results = []
+        if not self.value:
+            return results
+        separator = getattr(self, 'separator', ';')
+        uuids = self.value.split(separator)
+        if not isinstance(uuids, (list, tuple, set)):
+            uuids = [uuids]
```

Comment:
Am I missing something or `self.value.split(separator)` should always return a `list`?
With this assumption the check on line 637 looks redundant.

Reply:
true.